integration-tests/jest-reporter.js

```diff
@@ -0,0 +1,56 @@
+const _ = require('lodash/fp')
+const { parse } = require('path')
+const { createWriteStream, existsSync, mkdirSync } = require('fs')
+
+
+module.exports = class JestReporter {
+  constructor(_globalConfig, _options) {
+    this.logRootDir = process.env.LOG_DIR || '/tmp/test-results'
+  }
+
+  onTestStart(test) {
+    const { path } = test
+    console.info(`\n**  Running ${parse(path).name} at ${this.timeNow()}\n`)
+  }
+
+  onTestResult(_testRunConfig, testResult, _runResults) {
+    const { testFilePath, testResults } = testResult
+    const testName = parse(testFilePath).name
+    const hasFailure = testResults.some(({ status }) => status === 'failed')
+    const logDir = `${this.logRootDir}/${hasFailure ? 'failures' : 'successes'}`
+    const logFileName = `${logDir}/${testName}-${Date.now()}.log`
+    !existsSync(logDir) && mkdirSync(logDir, { recursive: true })
+
+    const writableStream = createWriteStream(logFileName)
+    writableStream.on('error', ({ message }) => {
+      console.error(`Error occurred while writing Console logs to ${logFileName}.\n${message}`)
+    })
+
+    _.forEach(({ message }) => writableStream.write(`${message}\n`), testResult.console)
+
+    // Write pass/failure summaries
+    writableStream.write('\n\nTests Summary\n')
+    _.forEach(result => {
+      writableStream.write('----------------------------------------------\n')
+      writableStream.write(`Test: ${result.title}\n`)
+      writableStream.write(`Status: ${result.status}\n`)
+      if (result.failureMessages.length > 0) {
+        const failure = result.failureMessages
+        writableStream.write(`Failure message: ${failure}\n`)
+      }
```

Comment:
How would you feel about

```suggestion
      const { failureMessages = [] } = result
      !_.isEmpty(failureMessages) && writableStream.write(`Failure message: ${failure}\n`)
```
?

Reply:
`failure` is undefined.
Updated to `!_.isEmpty(failureMessages) && writableStream.write(`Failure message: ${failureMessages}\n`)`